v2/api/dbformysql/v1beta1/user_types.go

```diff
@@ -0,0 +1,225 @@
+	return &genruntime.ResourceReference{
+		Group: group,
+		Kind:  kind,
+		Name:  user.Spec.Owner.Name,
+	}
+}
+
+// +kubebuilder:webhook:path=/validate-dbformysql-azure-com-v1beta1-user,mutating=false,sideEffects=None,matchPolicy=Exact,failurePolicy=fail,groups=dbformysql.azure.com,resources=users,verbs=create;update,versions=v1beta1,name=validate.v1beta1.users.dbformysql.azure.com,admissionReviewVersions=v1beta1
+
+var _ admission.Validator = &User{}
+
+// ValidateCreate validates the creation of the resource
+func (user *User) ValidateCreate() error {
+	validations := user.createValidations()
+	var temp interface{} = user
+	if runtimeValidator, ok := temp.(genruntime.Validator); ok {
+		validations = append(validations, runtimeValidator.CreateValidations()...)
+	}
+	var errs []error
+	for _, validation := range validations {
+		err := validation()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return kerrors.NewAggregate(errs)
+}
+
+// ValidateDelete validates the deletion of the resource
+func (user *User) ValidateDelete() error {
+	validations := user.deleteValidations()
+	var temp interface{} = user
+	if runtimeValidator, ok := temp.(genruntime.Validator); ok {
+		validations = append(validations, runtimeValidator.DeleteValidations()...)
+	}
+	var errs []error
+	for _, validation := range validations {
+		err := validation()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return kerrors.NewAggregate(errs)
+}
+
+// ValidateUpdate validates an update of the resource
+func (user *User) ValidateUpdate(old runtime.Object) error {
+	validations := user.updateValidations()
+	var temp interface{} = user
+	if runtimeValidator, ok := temp.(genruntime.Validator); ok {
+		validations = append(validations, runtimeValidator.UpdateValidations()...)
+	}
+	var errs []error
+	for _, validation := range validations {
+		err := validation(old)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return kerrors.NewAggregate(errs)
+}
+
+// createValidations validates the creation of the resource
+func (user *User) createValidations() []func() error {
+	return nil
+}
+
+// deleteValidations validates the deletion of the resource
+func (user *User) deleteValidations() []func() error {
+	return nil
+}
+
+// updateValidations validates the update of the resource
+func (user *User) updateValidations() []func(old runtime.Object) error {
+	return nil
+}
+
+// +kubebuilder:object:root=true
+type UserList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []User `json:"items"`
+}
+
+// TODO: Need to support rolling credentials
+
+type UserSpec struct {
+	//AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	//doesn't have to be.
+	AzureName string `json:"azureName,omitempty"` // TODO: Do we like this name? Or we want to use something else for handcrafted resources?
```

Comment:
I like the consistency of using the same name regardless of the resource.

Reply:
Removed TODO